src/content/drafts/custom-formatter-blog-post.md

```diff
@@ -0,0 +1,427 @@
+
+Next, add the [json2toml](https://www.npmjs.com/package/json2toml) package, which you'll use to help convert JavaScript objects to TOML: 
+
+```bash
+npm install json2toml
+```
+
+Now you're ready to write the custom formatter.
+
+### 2. Create the custom formatter 
+
+Custom formatters are just JavaScript functions that receive a `results` object and 
+optional `context` object as arguments and return a string as an output.
+In the custom formatter you'll make in this post, you'll only use the `results` argument.
+The `results` argument contains an array of all the ESLint results.
+For more information, refer to the [results object documentation](https://eslint.org/docs/latest/developer-guide/working-with-custom-formatters#the-result-object).
+
+In the `eslint-formatter-toml` directory, create a new file `formatter.js`.
+Now create the custom formatter in `formatter.js`:
+
+```js
+// Import package to convert JS objects to TOML
+const json2tomlConverter = require('json2toml');
+
+/**
+ * Custom formatter function that takes ESLint results and converts to TOML.
+ * @param {result[]} results See ESLint docs for more info on result object: 
+ *    https://eslint.org/docs/latest/developer-guide/working-with-custom-formatters#the-result-object 
+ * @returns {string} string representation of TOML results
+ */
+function formatTOML(results){
+  return json2tomlConverter(results);
+}
+
+// Exported custom formatter function
+module.exports = formatTOML;
+
+```
+
+### 3. Test the custom formatter
+
+With the custom formatter made, test it out locally. 
+
+First, you need the file that you're going to lint. 
+Add the directory `test-data` containing the file  `fullOfProblems.js`,
+which has some ESLint errors that you'll run the test on.
+Add the following contents to `test-data/fullOfProblems.js`:
+
+```js
+function addOne(i) {
+    if (i != NaN) {
+        return i ++
+    } else {
+      return
+    }
+};
+```
+
+Next, test your custom formatter out manually with the `eslint` CLI. 
+Include the path to your formatter as the argument to the `--format` flag.
+
+Run the following:
+
+```bash
+npx eslint test-data/fullOfProblems.js --format ./formatter.js
+```
+
+You should see a TOML version of the ESLint report in your terminal. 
+
+Now that the custom formatter is working, test it programmatically with a 
+unit test.
+Install the JavaScript testing package [Jest](https://jestjs.io/)
+and the [toml](https://www.npmjs.com/package/toml) package,
+which you'll use to test the custom formatter:
+
+```bash
+npm install jest toml --save-dev
+```
+
+In your `package.json` file, update your test script to use Jest:
+
+```json
+  // ...other config
+  "scripts": {
+    "test": "jest"
+  },
+  // ...other config
+```
+
+Also add support for Jest to your `.eslintrc.js` file:
+
+```js
+module.exports = {
+    "env": {
+        // other env configuration
+        "jest": true;
```

Comment:
Minor syntax issue that I noticed while following the example: a semicolon won't be expected here. Either no punctuation or a comma would work.

```suggestion
        "jest": true
```